pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -205,6 +207,21 @@ public void accept(Notification n) {
                 });
     }
 
+    @Override
+    public void registerSessionListener(Consumer<SessionEvent> listener) {
+        sessionListeners.add(listener);
+    }
+
+    protected void receivedSessionEvent(SessionEvent event) {
+        sessionListeners.forEach(l -> {
+            try {
+                l.accept(event);
+            } catch (Throwable t) {
+                log.warn("Error in processing session event", t);
```

Comment:
InterruptedException?

Reply:
The `Consumer` interface doesn't allow to throw `InterruptedException` and we cannot catch that directly since it's a checked exception. If the listener code does any blocking operation, it would be its responsibility to handle the `InterruptedException`.